pkg/resource/layer_version/hooks.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package layer_version
+
+import (
+	"context"
+	"sort"
+
+	// "github.com/Jeffail/gabs/v2"
+	svcapitypes "github.com/aws-controllers-k8s/lambda-controller/apis/v1alpha1"
+	svcsdk "github.com/aws/aws-sdk-go/service/lambda"
+)
+
+var (
+	_ = &svcsdk.Lambda{}
+	_ = &svcapitypes.LayerVersion{}
+)
+
+// customPreDelete deletes all the previous versions of a
+//LayerVersion except the latest version
+
+// This function is used as a sdk_delete hook, to delete
+// all the previous versions of a LayerVersion when delete API call is made
+func customPreDelete(
+	r *resource,
+	rm *resourceManager,
+	ctx context.Context,
+) error {
+	var response *svcsdk.ListLayerVersionsOutput
+	var err error
+	var version_list []*svcsdk.LayerVersionsListItem
+	var totalVersion int
+	var _ *svcsdk.DeleteLayerVersionOutput
```

Comment:
What is this needed for?

Reply:
The variables need to be declared before with their respective types, else they are creating error after in the code.

Reply:
That happens when you use `=` for declaration, if you use `:=` you don't need to above